### ISSUE TITLE
Repeat survey statement and set media group

### ISSUE THREAD

**Issue by MyZik** [php-telegram-bot/core#771]
Hi!

I want to build a 2-step survey command.

Step 1:
`Bot:` send me pictures and select "Done" button;
`User:` sends 1 photo
`User:` sends 1 photo
`User:` sends album with 4 photos
`User:` sends 1 photo 
`User:` **selects Done button**

Step 2:
`Bot:` **collects all the photos to new album**
`Bot:` "Done! Here is your album."
`Bot` **sends album**

Is it possible? I really don't know, how can I write this.

Thanks in advance!

**Comment by noplanman** [comment]
Something similar was asked in https://github.com/php-telegram-bot/core/issues/215#issuecomment-224381608

You should be able to use the code from there. It's not a pretty solution at the moment, but it works.

You can basically stay in the same state while the user sends photos, then change when a text (e.d. "done", maybe via keyboard) is sent.

**Comment by MyZik** [comment]
@noplanman thanks for your answer!
I tried to use that and I cannt get to the next step, after photo uploads :(

<offtop>And second question... How can I set the caption (Step 1) for all media group?</offtop>

```php
...

public function execute()
    {
        ...
        $result = Request::emptyResponse();
        $photos = [];
        $count = 0;

        switch ($state) {
            case 0:
                if ($message->getPhoto() === null || $text !== 'Закончить') {
                    $notes['state'] = 0;
                    $this->conversation->update();

                    /**
                     * Keyboard
                     */
                    $keyboard = new Keyboard(
                        ['Done']
                    );

                    $data['text'] = 'Upload the photo';

                    if ($text !== '') {
                        $data['text'] = 'You need to upload a photo or select "Done" button';
                    }

                    $data['reply_markup'] = $keyboard->setResizeKeyboard(true)->setOneTimeKeyboard(true)->setSelective(true);

                    if ($text === 'Done') {
                        $notes['state'] = 1;
//                        $this->conversation->update();
                        $data['reply_markup'] = Keyboard::remove();

                        break;
                    }

                    $result = Request::sendMessage($data);
                    break;
                }

                $photos .= new InputMediaPhoto([
                    'chat_id' => $chat_id,
                    'media' => $message->getPhoto()[$count]->getFileId(),
                ]);
                $notes['notes_' . $count . '_photo'] = $photos;
                
                $this->setObjectProperty($message, 'photo', null);
                
                $text = '';

            case 1:
                if ($text === '') {
                    $notes['state'] = 1;
                    $this->conversation->update();

                    $data['text'] = 'Write description of photos';
                    $result = Request::sendMessage($data);
                    break;
                }

                $notes['description'] = $text;
                $text = '';

            case 2:
                $this->conversation->update();
                unset($notes['state']);
                
                $data['text'] = 'Фотографии';

                $data['reply_markup'] = Keyboard::remove();

                $this->conversation->stop();
                $result = Request::sendMediaGroup(compact($notes['photos']));
                
                break;
        }
        return $result;
    }
```

**Comment by noplanman** [comment]
@MyZik If you have questions about the API, best to just check the official documentation first!
https://core.telegram.org/bots/api#inputmedia

Also, did you read the issue and comments I linked in my first reply to you?
It should explain the problem of sending the same message type multiple times in a survey.

**Comment by MyZik** [comment]
@noplanman 
When I copy an example and I've get an error: `PHP Fatal error:  Uncaught Error: Call to a member function getFileId() on null`;

Sorry for my stupid questions, I've updated my first post:

Step 1:
`Bot:` send me pictures and select "Done" button;
`User:` sends 1 photo
`User:` sends 1 photo
`User:` sends album with 4 photos
`User:` sends 1 photo 
`User:` **selects Done button**

Step 2:
`Bot:` **collects all the photos to new album**
`Bot:` "Done! Here is your album."
`Bot` **sends album**

I really don't know how can make this and see two problems...
1. How can I jump to a second step, when user selected "Done" button?
2. How collect all the photos to a one album and set it on `$notes['photos']` variable? 

Can you show me an example for my question..?
Thanks a lot in advance...

**Comment by noplanman** [comment]
Hi @MyZik 
I really can't be coding out commands like this for everyone 😁 

Just have another look at my previously linked issues and comments. All that you need for this should be there.

**Comment by R0of** [comment]
Really i need too.
noplanman - please show some example code 
problem not solved with: Something similar was asked in #215 (comment);
i need receive 2 or more photos in survey (by FileID!) and send it as Mediagroup;
(For "caption" in Mediagroup i find solution - i attach "caption" to first "Photo" and it's working.)
